### PR TITLE
fix(styles): make setting header UI better

### LINKED ISSUE #3055: New Settings header UI tweaks
Let's make the Avatar a bit bigger and make the overall layout mirror other pages a bit more closely.
@ckarlof @ryanfeeley 
settings

### PR DESCRIPTION
**Pull request by johngruen**
@pdehaan @zaach maybe give this a review? I added some stuff into the settings view to twiddle the DOM when a user adds or removes a display name.

Fixes #3055

### CODE REVIEW

**Comment by johngruen** [comment]
Fixes #3055

File: app/styles/_variables.scss [hunk 1]
```diff
@@ -35,6 +35,7 @@ $input-background-disabled-color: #f2f2f2;
 $input-border-disabled-color: #dddddd;
 $input-disabled-color: #64778c;
 $secondary-button-border-color: #c1c1c1;
+$black: #fff;
```
Comment (pdehaan) [inline app/styles/_variables.scss]: nit: this is a bit confusing. Since `$black` is technically white.

Comment (vladikoff) [inline app/styles/_variables.scss]: should this be $modal-border-color?

File: app/scripts/views/settings.js [hunk 2]
```diff
@@ -156,6 +159,25 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       });
     },
 
+    _swapDisplayName: function () {
+      var account = this.getSignedInAccount();
+      var displayName = account.get('displayName');
+      var email = account.get('email');
+
+      var $cardHeader = $('.card-header');
```
Comment (vladikoff) [inline app/scripts/views/settings.js]: could we use ` this.$...` for all these selectors here? see other usage in this same file

File: app/scripts/views/settings.js [hunk 3]
```diff
@@ -117,7 +117,10 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
         $.modal.close();
       }
       this.showEphemeralMessages();
+
```
Comment (vladikoff) [inline app/scripts/views/settings.js]: nit: extra spaces, not needed

**Comment by vladikoff** [comment]
image

Issues with the main header, letters get cut off

File: app/scripts/templates/settings.mustache [hunk 4]
```diff
@@ -1,24 +1,30 @@
...
   <header id="fxa-settings-header">
-    <h1 id="fxa-manage-account">{{#t}}Manage account{{/t}}</h1>
+    <h1 id="fxa-manage-account">{{#t}}Firefox Account{{/t}}</h1>
     {{#showSignOut}}
     <button id="signout" class="settings-button secondary">{{#t}}Sign out{{/t}}</button>
     {{/showSignOut}}
   </header>
+
   <div class="settings-success-wrapper">
     <div class="success settings-success">Hello</div>
```
Comment (vladikoff) [inline app/scripts/templates/settings.mustache]: Do we still need this `Hello` here?
1. it seems like it is not visible
2. the contents of that get replaced with some "updated" message. 

So should this just be ` <div class="success settings-success"></div>` ?

File: app/scripts/views/settings.js [hunk 5]
```diff
@@ -156,6 +159,25 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       });
     },
 
+    _swapDisplayName: function () {
```
Comment (shane-tomlinson) [inline app/scripts/views/settings.js]: @johngruen - can you document what this does and why it does it?

File: app/styles/_variables.scss (hunk 1 above)
Comment (johngruen) [inline app/styles/_variables.scss]: @pdehaan that IS confusing

File: app/scripts/views/settings.js (hunk 5 above)
Comment (johngruen) [inline app/scripts/views/settings.js]: @shane-tomlinson good idea

**Comment by johngruen** [comment]
@vladikoff so i fixed everything but am unable to repro the text clipping issue in FF or Chrome. Where were did you see this?

**Comment by vladikoff** [comment]
@johngruen Seems like text clipping was fixed but now the headers are just too close to each other: 

image

**Comment by vladikoff** [comment]
@johngruen the above is in latest stable chrome ^

**Comment by vladikoff** [comment]
It seems fine in the "small" view though, a bit close

**Comment by vladikoff** [comment]
Also needs rebase

File: app/scripts/views/settings.js [hunk 6]
```diff
@@ -148,6 +149,34 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       return self._showAvatar();
     },
 
+    beforeDestroy: function () {
+      $('body.settings').fadeOut(this.FADE_OUT_SETTINGS_MS, function (){
+        $('body').removeClass('settings').show();
+      });
+    },
```
Comment (vladikoff) [inline app/scripts/views/settings.js]: I think this change will introduce a flash of content, see gif:
flasg

@zaach ideas?

File: app/scripts/views/settings.js [hunk 7]
```diff
@@ -148,6 +149,34 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
...
+      if (displayName === cardHeader.text()) {
+        return;
+      } else if (! displayName) {
+        cardHeader.text(email);
+        cardSubheader.text('');
+      } else {
+        cardHeader.text(displayName);
+        cardSubheader.text(email);
+      }
+    },
```
Comment (vladikoff) [inline app/scripts/views/settings.js]: above is not covered by unit tests, see 


https://coveralls.io/builds/3670185/source?filename=app%2Fscripts%2Fviews%2Fsettings.js

Let us know if you need help adding tests for these

File: app/scripts/views/settings.js (hunk 6 above)
Comment (zaach) [inline app/scripts/views/settings.js]: Am I going crazy I thought we already had code that did this.

Comment (zaach) [inline app/scripts/views/settings.js]: Ugh, I didn't notice in https://github.com/mozilla/fxa-content-server/commit/207dfd69db6d34d882e3e1744d7d36aa42a42b83 that we removed the smooth transition.

/cc @shane-tomlinson

Comment (zaach) [inline app/scripts/views/settings.js]: This effect is now implemented in `base.js`, which we could modify to also use a transition if the view specifies a duration.

Comment (shane-tomlinson) [inline app/scripts/views/settings.js]: > Ugh, I didn't notice in 207dfd6 that we removed the smooth transition.
>
> /cc @shane-tomlinson

Yeah, that transition messed with the removal of the `layoutClassName`, so I took it out.

Comment (shane-tomlinson) [inline app/scripts/views/settings.js]: > Yeah, that transition messed with the removal of the layoutClassName, so I took it out.

I now see not having it messes with other things. In base.js, can you document why the transition is needed or else another drive by removal could occur.

File: app/scripts/views/settings.js [hunk 8]
```diff
@@ -148,6 +149,34 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
...
+      var cardHeader = this.$('.card-header');
+      var cardSubheader = this.$('.card-subheader');
+
+      if (displayName === cardHeader.text()) {
+        return;
+      } else if (! displayName) {
+        cardHeader.text(email);
+        cardSubheader.text('');
+      } else {
+        cardHeader.text(displayName);
```
Comment (shane-tomlinson) [inline app/scripts/views/settings.js]: This line and the next are not covered by unit tests:

screen shot 2015-10-01 at 11 45 29
.

File: app/tests/spec/views/settings.js [hunk 9]
```diff
@@ -521,6 +520,52 @@ function (chai, $, sinon, Cocktail, View, BaseView, SubPanels,
...
+
+              view._swapDisplayName();
+
+              assert.equal(view.$('.card-header').text(), 'a@a.com');
+              assert.equal(view.$('.card-subheader').text(), '');
+            });
+        });
+      });
+
+      describe('with display name added', function () {
```
Comment (johngruen) [inline app/tests/spec/views/settings.js]: @shane-tomlinson doesn't this cover that?

File: app/scripts/views/settings.js (hunk 6 above)
Comment (shane-tomlinson) [inline app/scripts/views/settings.js]: This flash still exists - can we just remove the fade out for now and file a follow on bug to re-add the behavior more generically?

File: app/tests/spec/views/settings.js [hunk 10]
```diff
@@ -521,6 +520,52 @@ function (chai, $, sinon, Cocktail, View, BaseView, SubPanels,
...
+
+          return view.render()
+            .then(function () {
+              view._swapDisplayName();
+
+              assert.equal(view.$('.card-header').text(), 'test user');
+              assert.equal(view.$('.card-subheader').text(), 'a@a.com');
+            });
+        });
+      });
```
Comment (shane-tomlinson) [inline app/tests/spec/views/settings.js]: As noted [above](https://github.com/mozilla/fxa-content-server/pull/3089/files#r40900440), one more test is needed that starts with the email address in the header, then adds a display name after the initial render. That should hit the uncovered code.

**Comment by johngruen** [comment]
@shane-tomlinson added another test that should fit the bill, also removed that fading stuff.

File: app/scripts/views/settings.js [hunk 11]
```diff
@@ -148,6 +149,32 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       return self._showAvatar();
     },
 
+    beforeDestroy: function () {
```
Comment (shane-tomlinson) [inline app/scripts/views/settings.js]: No need for this, it's done [automatically over in base.js](https://github.com/mozilla/fxa-content-server/blob/16c9fd54962e44292ec94a4305399bcacfa2d3cb/app/scripts/views/base.js#L373-L375)

File: app/scripts/templates/settings.mustache [hunk 12]
```diff
@@ -1,24 +1,30 @@
 <div id="fxa-settings-header-wrapper">
   <header id="fxa-settings-header">
-    <h1 id="fxa-manage-account">{{#t}}Manage account{{/t}}</h1>
+    <h1 id="fxa-manage-account">{{#t}}Firefox Account{{/t}}</h1>
```
Comment (shane-tomlinson) [inline app/scripts/templates/settings.mustache]: Do you want `Account` or `Accounts` here. I understand the rationale for `Account`, but that doesn't seem right.

Comment (ryanfeeley) [inline app/scripts/templates/settings.mustache]: I prefer "Accounts". It fits in more with the domain and the emails we are sending. You are at a site that has accounts, and maybe one day you'll be able to be signed into multple accounts, like Google.

**Comment by shane-tomlinson** [comment]
As soon as the tests pass, I'm hitting the button.

This is a great update @johngruen!

**Comment by johngruen** [comment]
cool!

**Comment by vladikoff** [comment]
r- 

display name evaluates javascript: 

image

**Comment by shane-tomlinson** [comment]
> display name evaluates javascript: 

oh dear lord. Great catch.

**Pull request merged**